### ISSUE TITLE
Inconsistent Timeout on CloudFront distribution creation (150+ distros)

### ISSUE THREAD

**Issue by Djiit** [terraform-providers/terraform-provider-aws#6197]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.40.0

### Affected Resource(s)

* aws_cloudfront_distribution

### Terraform Configuration Files

100 CF distributions with 50 rules each.

### Debug Output

```
* aws_cloudfront_distribution.xxx.1: error updating CloudFront Distribution (xxx): timeout while waiting for state to become 'success' (timeout: 1m0s)
(...)
```
(20x or so, mileage varies).

### Expected Behavior

Terraform should have applied our changes to AWS. (here, create 100-ish CloudFront (CF) distributions). Also we didn't expect a "1 minute timeout" on this resource. According to https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_cloudfront_distribution.go, it's 70 minute for CF distributions

### Actual Behavior

Terraform errored and leave multiple CF distros on AWS, forcing us to clean all this mess by hand.

### Steps to Reproduce

1. Create a .tf file with approx. 100 CF distributions
2. `terraform apply`

**Comment by bflad** [comment]
Hi @Djiit 👋 Very sorry for the trouble! Would it be possible to share what change(s) you were attempting? The `terraform plan` output would be very helpful in troubleshooting. If you also potentially have the [debug logging from Terraform](https://www.terraform.io/docs/internals/debugging.html), that might help determine if the timeout was occurring because of the AWS Go SDK automatically retrying or not. Its likely that when so many changes are occurring at once that the CloudFront API is throttling the requests in some fashion so we may need to tweak the logic around this.

That error message (with its 1 minute timeout for retries) occurs [here in the code](https://github.com/terraform-providers/terraform-provider-aws/blob/766b8ee9c94d4ae059ef77410b26a2d5078e44a4/aws/resource_aws_cloudfront_distribution.go#L802-L816), which is called during updates or during deletion (to disable the distribution first).

**Comment by Djiit** [comment]
I'm currently destroying theses CF distros (might have to wait for a couple hours), but i'll send you that as soon as I can. The `plan`output is *huge* (100*50 changes to apply).

**Comment by bflad** [comment]
If its giving you lots of trouble, you can also try reducing your concurrency with the terraform plan/apply [`-parallelism` flag](https://www.terraform.io/docs/commands/plan.html#parallelism-n) -- it defaults to 10. This will obviously slow down the process, but it might workaround it not working at all or leaving behind resources.

**Comment by Djiit** [comment]
Hmmm the plan output encoding is weird. I'll try again tomorrow.

**Comment by bflad** [comment]
Feel free to drop a [Gist](https://gist.github.com) which can be encrypted with the [HashiCorp GPG Key](https://keybase.io/hashicorp).

**Comment by Djiit** [comment]
Thanks, i'll do that. Some new information here : 

- When I try to re-apply, it tells me it needs to create 21 resources out of 161 (CF distributions) (like it doesn't know there are created), the same resources that errored ("CNAMEAlreadyExists"). The "apply" then fail the the exact same 21 errors. I can ran this multiple times with the same effect.

**Comment by Djiit** [comment]
Here is the encrypted output : https://gist.github.com/Djiit/cd40c6ad858b3ffa797ae466a3adf734

**Comment by yardensachs** [comment]
I'm also getting the "CloudFront Distribution: CNAMEAlreadyExists" not sure why

**Comment by bflad** [comment]
Pull request submitted: https://github.com/terraform-providers/terraform-provider-aws/pull/7809

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.1.0 of the Terraform AWS Provider, likely in the next day or two.

**Comment by bflad** [comment]
This has been released in [version 2.1.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#210-march-07-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
